test/parallel/test-whatwg-url-constructor.js

```diff
@@ -0,0 +1,143 @@
+'use strict';
+const common = require('../common');
+const path = require('path');
+const url = require('url');
+const URL = url.URL;
+const URLSearchParams = url.URLSearchParams;
+const { test, assert_equals, assert_true, assert_throws } = common.WPT;
+
+if (!common.hasIntl) {
+  // A handful of the tests fail when ICU is not included.
+  common.skip('missing Intl');
+  return;
+}
+
+const request = {
+  response: require(path.join(common.fixturesDir, 'url-tests.json'))
+};
+
+/* eslint-disable */
+/* WPT Refs:
+   https://github.com/w3c/web-platform-tests/blob/8791bed/url/url-constructor.html
+*/
+function runURLConstructorTests() {
+  // var setup = async_test("Loading data…")
+  // setup.step(function() {
+  //   var request = new XMLHttpRequest()
+  //   request.open("GET", "urltestdata.json")
+  //   request.send()
+  //   request.responseType = "json"
+  //   request.onload = setup.step_func(function() {
+         runURLTests(request.response)
+  //     setup.done()
+  //   })
+  // })
+}
+
+function bURL(url, base) {
+  return new URL(url, base || "about:blank")
+}
+
+
+function runURLTests(urltests) {
+  for(var i = 0, l = urltests.length; i < l; i++) {
+    var expected = urltests[i]
+    if (typeof expected === "string") continue // skip comments
+
+    test(function() {
+      if (expected.failure) {
+        assert_throws(new TypeError(), function() {
+          bURL(expected.input, expected.base)
+        })
+        return
+      }
+
+      var url = bURL(expected.input, expected.base)
+      assert_equals(url.href, expected.href, "href")
+      assert_equals(url.protocol, expected.protocol, "protocol")
+      assert_equals(url.username, expected.username, "username")
+      assert_equals(url.password, expected.password, "password")
+      assert_equals(url.host, expected.host, "host")
+      assert_equals(url.hostname, expected.hostname, "hostname")
+      assert_equals(url.port, expected.port, "port")
+      assert_equals(url.pathname, expected.pathname, "pathname")
+      assert_equals(url.search, expected.search, "search")
+      // if ("searchParams" in expected) {
```

Comment:
`"??a=b&c=d"` is broken at the moment, can make another PR to skip specific test cases later I think (though the whole condition doesn't have to be commented out. I'll change that)